### PR TITLE
Add Say() support for Sonos devices (initial contribution by Jochen H…

### PR DESCRIPTION
**Pull request by kgoderis**
…iller) + fix for Google Captchas

Signed-off-by: Karel Goderis <karel.goderis@me.com>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.multimedia.sonos/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ org.eclipse.smarthome.config.core,
+ org.eclipse.smarthome.core.items,
+ org.eclipse.smarthome.core.library.items,
+ org.eclipse.smarthome.core.library.types,
+ org.eclipse.smarthome.core.scriptengine,
+ org.eclipse.smarthome.core.scriptengine.action,
+ org.eclipse.smarthome.io.multimedia.tts,
+ org.osgi.framework,
+ org.osgi.service.component;version="1.2.2",
+ org.osgi.service.http;version="1.2.1",
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.multimedia.sonos/META-INF/MANIFEST.MF]: Do we need all that version contraints?
Does it not work with servlet < 2.6.0, SCR < 1.2.2?

File: bundles/io/org.eclipse.smarthome.io.multimedia.sonos/OSGI-INF/multimedia_sonos.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+-->
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.eclipse.smarthome.io.multimedia.tts.googletranslate.tts" immediate="true">
+   <implementation class="org.eclipse.smarthome.io.multimedia.sonos.internal.MultimediaTTSServiceSonos"/>
+   <service>
+      <provide interface="org.eclipse.smarthome.io.multimedia.tts.TTSService"/>
+   </service>
+   <property name="os" type="String" value="macosx"/>
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.multimedia.sonos/OSGI-INF/multimedia_sonos.xml]: What is the intention of the "os" property? Does it work on macosx only?

File: bundles/io/org.eclipse.smarthome.io.multimedia.sonos/OSGI-INF/multimedia_sonos.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+   <property name="service.ranking" type="Integer" value="10000"/>
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.multimedia.sonos/OSGI-INF/multimedia_sonos.xml]: Could we set the service ranking by the configuration admin or have it be fix?

**Comment by maggu2810** [comment]
IMHO we should not bundle a log implementation (logback).

File: bundles/io/org.eclipse.smarthome.io.multimedia.sonos/pom.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  </parent>
+
+  <properties>
+    <bundle.symbolicName>org.eclipse.smarthome.io.multimedia.tts.googletranslate</bundle.symbolicName>
+    <bundle.namespace>org.eclipse.smarthome.io.multimedia.tts.googletranslate</bundle.namespace>
+  </properties>
+
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.eclipse.smarthome.io</groupId>
+  <artifactId>org.eclipse.smarthome.io.multimedia.tts.googletranslate</artifactId>
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.multimedia.sonos/pom.xml]: Shouldn't the artifact ID be similar to the directory (org.eclipse.smarthome.io.multimedia.tts.googletranslate vs. org.eclipse.smarthome.io.multimedia.sonos)?

**Comment by maggu2810** [comment]
IMHO you should also drop the logback.xml.

File: bundles/io/org.eclipse.smarthome.io.multimedia.sonos/pom.xml (hunk 4 above)
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.multimedia.sonos/pom.xml]: Same for artifact ID == BSN

**Comment by kaikreuzer** [comment]
@maggu2810: Before you continue:
@kgoderis: I am not sure if we should add this to openHAB 2 right now - after all, it is already in ESH namespace, so it cannot be merged here anyhow.
Let us rather look together with @JochenHiller, how we would want to add this feature to ESH, once the Sonos binding is there, ok?

**Comment by kgoderis** [comment]
yeah - I completely misread the package naming and was off-guarded by the fact that sonos is still in OH2. No prob to do that in ESH. the code works, I verified this in my setup. Already talked to Jochen a few weeks a go to get it integrated but I guess he is busy, hence the initiative. Maybe then Jochen can also quickly integrate the PR discussion feedback?

**Comment by kgoderis** [comment]
The code comes from a diff dating back September 2014! So it might need some refreshment

**Comment by kaikreuzer** [comment]
> Maybe then Jochen can also quickly integrate the PR discussion feedback?

Haha, yes, I am sure he will :-)

**Comment by kgoderis** [comment]
That sound pretty sarcastic ;-) I am missing something here? ;-) ;-)

**Comment by kaikreuzer** [comment]
No, not at all. It just seemed to be a good opportunity to run away from @maggu2810 review comments 8-)

**Comment by kgoderis** [comment]
Well - he *is* thorough :/) but that is probably something common to most Germans - no pun intended :-) :-) pfewwww I am happy I only fixed the Google API part here

**Pull request closed without merging**